### PR TITLE
Add eslint-plugin-react-with-styles, add root: true to .eslintrc

### PR DESCRIPTION
**Pull request by lencioni**
These eslint rules will help protect this project from bad
react-with-styles usage.

@majapw

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -8,8 +11,13 @@
 
   "rules": {
     "no-mixed-operators": [2, { "allowSamePrecedence": true }],
+
     "jsx-a11y/no-static-element-interactions": 1, // TODO: enable
+
     "react/no-did-mount-set-state": 0, // necessary for server-rendering
+
+    "react-with-styles/no-unused-styles": 2,
```
Comment (ljharb) [inline .eslintrc]: do these come in with a recommended config at all, or only enabled one at a time?

Comment (lencioni) [reply .eslintrc]: One of them is in recommended config. The other is not yet.

Comment (ljharb) [reply .eslintrc]: Should we use the recommended config then, and only add the rule that differs?

Comment (lencioni) [reply .eslintrc]: Sure thing, I'll push an update shortly.

**Pull request merged**